python/cudf/cudf/core/indexed_frame.py

```diff
@@ -3308,9 +3308,13 @@ def _split(self, splits, keep_index: bool = True) -> list[Self]:
             splits,
         )
 
+        @acquire_spill_lock()
+        def split_from_pylibcudf(split: list[plc.Column]) -> list[ColumnBase]:
+            return [ColumnBase.from_pylibcudf(col) for col in split]
+
```

Comment:
Does this work?
```suggestion
        with acquire_spill_lock()
            columns = [ColumnBase.from_pylibcudf(col) for col in split]
```

Reply:
`split` here is an iterator variable (from `for split in columns_split`) below.

So in order to do it like this I would have to do 

```python
with acquire_spill_lock():
    column = [[ColumnBase.from_pylibcudf(col) for col in split] for split in columns_split]
```

I could put the entire operation in the `with acquire_spill_lock()` block, but I would prefer tightly scope the operation that definitely needs the spill lock.